Keras convert dense for n-D input is not correct.

In the current test_layer.py:
```
    def test_dense(self):
        input_data = np.random.random_sample([2, 10, 5, 7])
        layer = Dense(2, init='one', activation="relu",
                      input_shape=(10, ), W_regularizer=l1l2(l1=0.01, l2=0.02))
        input_data = np.random.random_sample([2, 10])
        self.modelTestSingleLayer(input_data, layer, dump_weights=True)
        layer2 = Dense(2, init='one', activation="softplus",
                       input_shape=(10, ), b_regularizer=l2(0.02))
        self.modelTestSingleLayer(input_data, layer2, dump_weights=True)
        layer3 = Dense(2, init='one', input_shape=(10, ),
                       W_regularizer=keras.regularizers.WeightRegularizer(l1=0.1))
        self.modelTestSingleLayer(input_data, layer3, dump_weights=True)
        layer4 = Dense(2, init='glorot_uniform', activation="hard_sigmoid", input_shape=(10, ))
        self.modelTestSingleLayer(input_data, layer4, dump_weights=True)
```
The first line input_data has been overridden. Actually the case for nD input is never tested.

If I run the following:
```
        input_data = np.random.random_sample([2, 10, 5, 7])
        layer = Dense(2, init='one', activation="relu",
                      input_shape=(10, 5, 7), W_regularizer=l1l2(l1=0.01, l2=0.02))
        self.modelTestSingleLayer(input_data, layer, dump_weights=True)
```
Will get error:
```
E                   java.lang.IllegalArgumentException: requirement failed: size mismatch, m1:350x2 m2:7x2
E                   	at scala.Predef$.require(Predef.scala:224)
E                   	at com.intel.analytics.bigdl.tensor.DenseTensorMath$.addmm(DenseTensorMath.scala:515)
E                   	at com.intel.analytics.bigdl.tensor.DenseTensor.addmm(DenseTensor.scala:1230)
E                   	at com.intel.analytics.bigdl.nn.Linear.updateOutput(Linear.scala:108)
E                   	at com.intel.analytics.bigdl.nn.Linear.updateOutput(Linear.scala:45)
E                   	at com.intel.analytics.bigdl.nn.abstractnn.AbstractModule.forward(AbstractModule.scala:268)
E                   	at com.intel.analytics.bigdl.nn.Sequential.updateOutput(Sequential.scala:39)
E                   	at com.intel.analytics.bigdl.nn.abstractnn.AbstractModule.forward(AbstractModule.scala:268)
E                   	at com.intel.analytics.bigdl.nn.Sequential.updateOutput(Sequential.scala:39)
E                   	at com.intel.analytics.bigdl.nn.abstractnn.AbstractModule.forward(AbstractModule.scala:268)
E                   	at com.intel.analytics.bigdl.nn.StaticGraph.updateOutput(StaticGraph.scala:57)
E                   	at com.intel.analytics.bigdl.nn.abstractnn.AbstractModule.forward(AbstractModule.scala:268)
E                   	at com.intel.analytics.bigdl.python.api.PythonBigDL.modelForward(PythonBigDL.scala:1910)
```

--- a/pyspark/bigdl/keras/converter.py
+++ b/pyspark/bigdl/keras/converter.py
@@ -515,7 +515,7 @@ class LayerConverter:
             return self.combo_parameter_layer(blayer, self.config)
         else:
             seq = BLayer.Sequential()
-            seq.add(BLayer.InferReshape([-1, out_dim], False))
+            seq.add(BLayer.InferReshape([-1, in_dim], False))
             seq.add(blayer)
             seq.add(BLayer.InferReshape([-1] + list(self.input_shape[1:-1]) + [out_dim], False))
             return self.combo_parameter_layer(seq, self.config)

--- a/pyspark/test/bigdl/keras/test_layer.py
+++ b/pyspark/test/bigdl/keras/test_layer.py
@@ -68,17 +68,17 @@ class TestLayer(BigDLTestCase):
     def test_dense(self):
         input_data = np.random.random_sample([2, 10, 5, 7])
         layer = Dense(2, init='one', activation="relu",
-                      input_shape=(10, ), W_regularizer=l1l2(l1=0.01, l2=0.02))
-        input_data = np.random.random_sample([2, 10])
+                      input_shape=(10, 5, 7), W_regularizer=l1l2(l1=0.01, l2=0.02))
         self.modelTestSingleLayer(input_data, layer, dump_weights=True)
+        input_data2 = np.random.random_sample([2, 10])
         layer2 = Dense(2, init='one', activation="softplus",
                        input_shape=(10, ), b_regularizer=l2(0.02))
-        self.modelTestSingleLayer(input_data, layer2, dump_weights=True)
+        self.modelTestSingleLayer(input_data2, layer2, dump_weights=True)
         layer3 = Dense(2, init='one', input_shape=(10, ),
                        W_regularizer=keras.regularizers.WeightRegularizer(l1=0.1))
-        self.modelTestSingleLayer(input_data, layer3, dump_weights=True)
+        self.modelTestSingleLayer(input_data2, layer3, dump_weights=True)
         layer4 = Dense(2, init='glorot_uniform', activation="hard_sigmoid", input_shape=(10, ))
-        self.modelTestSingleLayer(input_data, layer4, dump_weights=True)
+        self.modelTestSingleLayer(input_data2, layer4, dump_weights=True)
 
     def test_timedistributeddense(self):
         input_data = np.random.random_sample([2, 4, 5])

